marklogic-spark-connector/src/main/java/com/marklogic/spark/udf/TextClassifierUdf.java

```diff
@@ -41,16 +44,37 @@ public static UserDefinedFunction build(
         additionalParameters.put("apiKey", apiKey);
         additionalParameters.put("tokenEndpoint", tokenEndpoint);
         classificationConfiguration.setAdditionalParameters(additionalParameters);
-        return functions.udf(TextClassifierUdf::classifyText, DataTypes.BinaryType);
+        return functions.udf(TextClassifierUdf::classifyText, DataTypes.createArrayType(DataTypes.StringType));
     }
 
-    private static byte[] classifyText(Object binaryContent) {
-        if (!(binaryContent instanceof byte[])) {
+    private static List<String> classifyText(Object binaryContent) {
+        // Need to add handler for when a token expires midstream.
+        if (binaryContent instanceof byte[]) {
+            fetchTokenIfNecessary();
+            List<String> classifications = new ArrayList<>();
+            String content = new String((byte[]) binaryContent, StandardCharsets.UTF_8);
```

Comment:
This is fine for now but may need to become an option, as we can't assume we have UTF-8 content from a file. I can add a test later that uses a non-UTF-8 file to see what happens.